src/rules/__tests__/unbound-method.test.ts

```diff
@@ -113,33 +113,37 @@ const invalidTestCases: Array<TSESLint.InvalidTestCase<MessageIds, Options>> = [
     ],
   },
   // toThrow matchers call the expected value (which is expected to be a function)
-  ...toThrowMatchers.map(matcher => ({
-    code: dedent`
+  ...toThrowMatchers.map<TSESLint.InvalidTestCase<MessageIds, Options>>(
+    matcher => ({
+      code: dedent`
       ${ConsoleClassAndVariableCode}
 
       expect(console.log).${matcher}();
     `,
-    errors: [
-      {
-        line: 9,
-        messageId: 'unboundWithoutThisAnnotation' as const,
-      },
-    ],
-  })),
+      errors: [
+        {
+          line: 9,
+          messageId: 'unboundWithoutThisAnnotation',
+        },
+      ],
+    }),
+  ),
   // toThrow matchers call the expected value (which is expected to be a function)
-  ...toThrowMatchers.map(matcher => ({
-    code: dedent`
+  ...toThrowMatchers.map<TSESLint.InvalidTestCase<MessageIds, Options>>(
```

Comment:
Ditto for this

Reply:
Got it. I've addressed it in commit https://github.com/jest-community/eslint-plugin-jest/commit/b030cc748361ecd7692845e22114be041700e9b3.